controllers/config_helper.go

```diff
@@ -0,0 +1,195 @@
+package controllers
+
+import (
+	"context"
+
+	operatorv1alpha1 "github.com/gatekeeper/gatekeeper-operator/api/v1alpha1"
+	"github.com/go-logr/logr"
+	"github.com/open-policy-agent/gatekeeper/v3/apis/config/v1alpha1"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/wildcard"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	cacheRuntime "sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/metrics/server"
+)
+
+// Default config data
+func getDefaultConfig(namespace string) *v1alpha1.Config {
+	config := &v1alpha1.Config{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      "config",
+		},
+		Spec: v1alpha1.ConfigSpec{
+			Match: []v1alpha1.MatchEntry{
+				{
+					ExcludedNamespaces: []wildcard.Wildcard{
+						"kube-*", "multicluster-engine",
+						"hypershift", "hive", "rhacs-operator", "open-cluster-*", "openshift-*",
+					},
+					Processes: []string{
+						"webhook", "mutation-webhook",
+					},
+				},
+			},
+		},
+	}
+
+	return config
+}
+
+func createDefaultConfig(ctx context.Context, c client.Client, namespace string,
```

Comment:
This should set the owner's reference on the Config object though.

Reply:
Just a question. Do we need this? when the gatekeeper(resource) is deleted, the config is also deleted too. Because CRD is deleted

Reply:
The owner's reference is how the operator will know it has ownership of the Config object and will always set `matches` based on what is/isn't in the `Gatekeeper` operator CR.

A `Config` object without an owner's reference with user provided `matches` on the `Config` object indicates it is a legacy configuration.